Originalidk iP

Originalidk opened the pull request:
#ChatGP0
> “Chatbots represent a new trend in how people access information, make decisions and communicate.”  ~Christie Pitts [\(Source\)](https://www.forbes.com/sites/christinecrandell/2016/10/23/chatbots-will-be-your-new-best-friend/?sh=592715f1a245)

ChatGP0 helps you keep track of what needs to be done. It is
- Text-based
- Easy to use
- Very _Fast_ 🥇

ChatGP0 keeps track of:
1. Todo Tasks
2. Tasks with deadlines
3. Events

**Features**
- [ ] Create tasks
- [ ] Delete tasks
- [ ] Mark/unmark tasks
- [ ] Find tasks

Here is the code to `run` the program
```
 public void run() {
        Ui.start();
        Scanner scan = new Scanner(System.in);
        String input = Ui.getInput(scan);
        while (!input.equals("bye")) {
            Parser.parseInput(input, this.tasks, this.storage);
            input = Ui.getInput(scan);
        }
        Ui.bye();
    }
```

Commit:
Add Gradle support

Eclipse-Dominator committed:
Bump gradle and lib version

Originalidk committed:
Level-0

Originalidk committed:
Level-1

Originalidk committed:
Level-2

Originalidk committed:
Level-3

Originalidk committed:
Level-4

Originalidk committed:
A-TextUiTesting

Originalidk committed:
Level-5

Originalidk committed:
Level-6

Originalidk committed:
Level-7-branch

Originalidk committed:
Merge branch 'branch-Level-7'

Originalidk committed:
Level-8 split fail

Originalidk committed:
Level-8 Deadline not parsing

Originalidk committed:
Level-8 Completed

Originalidk committed:
Merge branch 'branch-Level-8'

Originalidk committed:
OOP files no test

Originalidk committed:
OOP tests passed

Originalidk committed:
Merge branch 'branch-A-MoreOOP'

Originalidk committed:
A-Packages

Originalidk committed:
Delete comments

Originalidk committed:
Merge branch 'branch-A-Packages'

Originalidk committed:
Merge branch 'add-gradle-support' into branch-A-Gradle

Originalidk committed:
Add build.gradle

Originalidk committed:
Merge branch 'branch-A-Gradle'

Originalidk committed:
Add test cases

Originalidk committed:
Merge branch 'branch-A-JUnit'

Originalidk committed:
Add JAR file

Originalidk committed:
Merge branch 'branch-A-Jar'

Originalidk committed:
Add javadoc comments to non-test classes

Originalidk committed:
Follow coding standard

Originalidk committed:
Add find command

Originalidk committed:
Merge branch 'branch-A-CodingStandard'

Originalidk committed:
Add java docs with resolved conflicts

Originalidk committed:
Merge branch 'branch-level-9'

joel-foo commented on src/main/java/duke/Ui.java:
```diff
@@ -0,0 +1,158 @@
+     * Prints the starting messages of the chatbot.
+     */
+    public static void start() {
+        System.out.println("    ____________________________________________________________");
```
Maybe can create a constant for DIVIDER for 
`"    ____________________________________________________________"`

joel-foo commented on src/main/java/duke/DeadlineUnclearException.java:
I couldn't find where this exception is instantiated/used. Same for the other exceptions as well.

joel-foo commented on src/main/java/duke/Parser.java:
```diff
@@ -0,0 +1,133 @@
+     * @param tasks List of tasks.
+     * @param storage Stores the file and handles file methods.
+     */
+    public static void parseInput(String input, TaskList tasks, Storage storage) {
```
perhaps you may want to call .trim() on input first before the if else loops - to account for edge cases. For example if I key in "   list    " instead of "list".

joel-foo commented on src/main/java/duke/TaskList.java:
```diff
@@ -0,0 +1,152 @@
+ */
+public class TaskList {
+    /** List of tasks. **/
+    ArrayList<Task> list;
+    /** File storing information to create the list of tasks. **/
+    File startFile;
```
can make this `private`?

joel-foo commented on src/main/java/duke/Duke.java:
```diff
@@ -0,0 +1,44 @@
+            Parser.parseInput(input, this.tasks, this.storage);
+            input = Ui.getInput(scan);
+        }
+        Ui.bye();
```
remember to close the scanner.

perhaps you can bundle the scanner as part of the Ui class for better encapsulation.

-> you can instantiate the scanner in Ui. and when you call the bye function, you can close the scanner and print the goodbye message.

joel-foo commented on src/main/java/duke/Event.java:
```diff
@@ -0,0 +1,45 @@
+
+    @Override
+    public String toString() {
+        return "[E]" + super.toString() + " (from: " + this.startDateTime.format(DateTimeFormatter.ofPattern("MMM d yyyy HHmm"))
+                + " to: " + this.endDateTime.format(DateTimeFormatter.ofPattern("MMM d yyyy HHmm")) + ")";
```
perhaps can make a separate DatetimeParser class?

 there's many repeated usages of
`.format(DateTimeFormatter.ofPattern("MMM d yyyy HHmm")` in the Event and Deadline classes.

joel-foo commented on src/test/java/duke/DeadlineTest.java:
```diff
@@ -0,0 +1,35 @@
+public class DeadlineTest {
+
+    @Test
+    public void testStringConversion() {
```
you may want to name your test functions more explicitly - indicate whether you expect a success / an exception to be thrown.

joel-foo commented on src/test/java/duke/StorageTest.java:
```diff
@@ -0,0 +1,57 @@
+
+import static org.junit.jupiter.api.Assertions.fail;
+
+public class StorageTest {
```
I like that you have very comprehensive tests.

Originalidk committed:
Add Commands

Originalidk committed:
Merge branch 'branch-commands'

Originalidk committed:
Add checkstyle and fixed style issues

Originalidk committed:
Merge branch 'branch-checkstyle'

vijay-shankaranand commented on src/main/java/duke/EventCommand.java:
```diff
@@ -0,0 +1,43 @@
+        } catch (EmptyDescriptionException e) {
+            Ui.emptyDesc("event");
+        } catch (DurationUnclearException e) {
+            Ui.unclear("duration");
```
I like that you have caught possible exceptions neatly and display to your user using your UI component!

vijay-shankaranand commented on src/main/java/duke/Ui.java:
```diff
@@ -0,0 +1,164 @@
+     * Prints list of tasks that matches the search input.
+     *
+     * @param selectedTasks List of tasks.
+     */
```
Good job on the clean and concise documentation of all your methods!

vijay-shankaranand commented on src/test/java/duke/EventTest.java:
```diff
@@ -0,0 +1,40 @@
+import java.time.LocalDateTime;
+import java.time.format.DateTimeFormatter;
+
+import org.junit.jupiter.api.Test;
```
Perhaps you could consider grouping various package imports together
```suggestion
import org.junit.jupiter.api.Test;
import static org.junit.jupiter.api.Assertions.assertEquals;

import java.time.LocalDateTime;
import java.time.format.DateTimeFormatter;
```

vijay-shankaranand commented on src/test/java/duke/EventTest.java:
```diff
@@ -0,0 +1,40 @@
+        Task task = new Event("Make my resume", fromDateTime, toDateTime);
+        task.setMark(true);
+
+        assertEquals("[E][X] Make my resume (from: Sep 23 2023 1800 to: Sep 23 2023 2000)", task.toString());
```
I like how you have spaced out lines according to their function!

vijay-shankaranand commented on text-ui-test/input.txt:
```diff
@@ -0,0 +1,45 @@
+unmark 99
+delete 99
+delete 1
+delete 1
```
I like that you have very comprehensive test inputs!

vijay-shankaranand reviewed (commented):
Overall, an almost excellent attempt to follow the coding standard, with a few minor nits. LGTM otherwise

Originalidk committed:
Add incomplete GUI

Originalidk committed:
Add unlimited height for label

Originalidk committed:
Merge branch 'branch-GUI'

Originalidk committed:
Add comments for more methods and fields

Originalidk committed:
Add assert statements

There are several classes which rely on a txt file being created or
already existing.

Asserting if the file exists ensures that these classes work as intended.

Originalidk committed:
Improve quality of boolean conditions and comments

Boolean conditions might look complicated without explainations.
Comments in the code were also inaccurate for commands and Ui methods.

Assigning the boolean conditions to variables makes the code easier to
read. Correcting the comments prevents confusion from occuring.

Originalidk committed:
Merge pull request #1 from Originalidk/branch-A-Assertions

Add assert statements

Originalidk committed:
Revert "Add assert statements"

Originalidk committed:
Merge pull request #2 from Originalidk/revert-1-branch-A-Assertions

Revert "Add assert statements"

Originalidk committed:
Revert "Revert "Add assert statements""

Originalidk committed:
Merge pull request #3 from Originalidk/revert-2-revert-1-branch-A-Assertions

Revert "Revert "Add assert statements""

Originalidk committed:
Merge branch 'master' of https://github.com/Originalidk/ip into branch-A-CodeQuality

Originalidk committed:
Merge pull request #4 from Originalidk/branch-A-CodeQuality

Branch a code quality

Originalidk committed:
Add contact feature

Prior to this, the chatbot can only store tasks.

The contact feature allows users to store contacts.

Originalidk committed:
Merge pull request #5 from Originalidk/branch-D-Contacts

Add contact feature

Originalidk committed:
Add title and Ui.png

Originalidk committed:
Merge pull request #7 from Originalidk/branch-finalise-features

Add title and Ui.png

Originalidk committed:
Edit build file

Originalidk committed:
Add user guide

Originalidk committed:
Merge pull request #8 from Originalidk/branch-A-UserGuide

Add user guide

Originalidk committed:
Edit comment
